src/WebJobs.Extensions/Extensions/Core/Bindings/ExecutionContextBindingProvider.cs

```diff
@@ -63,10 +60,17 @@ public Task<IValueProvider> BindAsync(object value, ValueBindingContext context)
                     throw new ArgumentNullException("context");
                 }
 
-                return BindInternalAsync(new ExecutionContext
+                return BindInternalAsync(CreateContext(context.FunctionInstanceId));
+            }
+
+            private ExecutionContext CreateContext(Guid invocationId)
+            {
+                return new ExecutionContext
                 {
-                    InvocationId = context.FunctionInstanceId
-                });
+                    InvocationId = invocationId,
+                    FunctionName = _parameter.Member.Name,
```

Comment:
I think we have a few different ways of getting FunctionName in the core sdk (FullName, ShortName, etc) - is this compatible?  What's the intended usage?